db/migrate/20210120150019_add_position_to_ansible_role.rb

```diff
@@ -0,0 +1,31 @@
+class AddPositionToAnsibleRole < ActiveRecord::Migration[6.0]
+  def change
+    add_column :host_ansible_roles, :position, :integer
+    add_column :hostgroup_ansible_roles, :position, :integer
+
+    update_hostgroup_ansible_roles
+    update_host_ansible_roles
+    change_column_null :host_ansible_roles, :position, false
+    change_column_null :hostgroup_ansible_roles, :position, false
+  end
+
+  def update_host_ansible_roles
+    hosts = Host.unscoped.where(:id => HostAnsibleRole.distinct.pluck(:host_id).uniq)
+    hosts.each do |host|
+      host.host_ansible_roles.each_with_index do |har, idx|
+        har.position = idx
+        har.save!
+      end
+    end
```

Comment:
This is highly inefficient :worried: What about something like: ?
```suggestion
    HostAnsibleRole.all.pluck(:host_id, :id).group_by(&:first) do |host_id, role_ids|
      role_ids.each_with_index do |(host_id, host_role_id), idx|
        HostAnsibleRole.where(id: host_role_id).update_all(position: idx)
      end
    end
```